src/scripts/OSFramework/OSUI/Feature/Balloon/Balloon.ts

```diff
@@ -0,0 +1,332 @@
+				// Remove FloatingUI
+				this._floatingUIInstance.close();
+				// Focus on last element clicked. Async to avoid conflict with closing animation
+				Helper.AsyncInvocation(() => {
+					this.featureElem.blur();
+					if (isBodyClick === false) {
+						this._focusableActiveElement.focus();
+					}
+				});
+			}
+
+			// Trigger the Custom Event BalloonOnToggle
+			this._onToggleEvent(this.isOpen, this.featureElem);
+
+			// Delay the _isOpenedByApi assignement in order to deal with clickOnBody() and open() api concurrency!
+			Helper.AsyncInvocation(() => {
+				this._isOpenedByApi = false;
+			});
+		}
+
+		// Method to unset the callbaks
+		private _unsetCallbacks(): void {
+			this._eventBodyClick = undefined;
+			this._eventOnKeypress = undefined;
+			this._onToggleEvent = undefined;
+			window[OSFramework.OSUI.GlobalEnum.CustomEvent.BalloonOnToggle] = undefined;
+		}
+
+		/**
+		 * Method to build the Feature
+		 *
+		 * @memberof Balloon
+		 */
+		public build(): void {
+			this._setCallbacks();
+			this._setEventListeners();
+			this.setFloatingUIBehaviour();
+			this._handleFocusTrap();
+			this._setA11YProperties();
+			this.setBalloonShape();
+		}
+
+		/**
+		 * Method to close the Balloon
+		 *
+		 * @memberof Balloon
+		 */
+		public close(): void {
+			if (this.isOpen) {
+				this._toggleBalloon(false);
+			}
+		}
+
+		/**
+		 * Destroy the Balloon.
+		 *
+		 * @memberof Balloon
+		 */
+		public dispose(): void {
+			this._floatingUIInstance.dispose();
+			this._unsetCallbacks();
+			super.dispose();
+		}
+
+		/**
+		 * Method to open the Balloon
+		 *
+		 * @memberof Balloon
+		 */
+		public open(isOpenedByApi: boolean): void {
+			if (this.isOpen === false) {
+				this._isOpenedByApi = isOpenedByApi;
+				this._toggleBalloon(true);
+			}
+		}
+
+		/**
+		 * Method to handle the Shape config css variable
+		 *
+		 * @param {GlobalEnum.ShapeTypes} [shape]
+		 * @memberof Balloon
+		 */
+		public setBalloonShape(shape?: GlobalEnum.ShapeTypes): void {
+			if (shape !== undefined) {
+				this.featureOptions.shape = shape;
+			}
+
+			Helper.Dom.Styles.SetStyleAttribute(
+				this.featureElem,
+				Enum.CssCustomProperties.Shape,
+				'var(--border-radius-' + this.featureOptions.shape + ')'
+			);
+		}
+
+		/**
+		 * Method to set the FloatingUI provider instance
+		 *
+		 * @param {boolean} [isUpdate]
+		 * @memberof Balloon
+		 */
+		public setFloatingUIBehaviour(isUpdate?: boolean): void {
+			if (isUpdate || this._floatingUIInstance === undefined) {
+				this.setFloatingUIOptions();
+
+				if (isUpdate && this._floatingUIInstance !== undefined) {
+					this._floatingUIInstance.update(this._floatingUIOptions);
+				}
+
+				this._floatingUIInstance = new Providers.OSUI.Utils.FloatingUI(this._floatingUIOptions);
+			} else {
+				this._floatingUIInstance.build();
+			}
+		}
+
+		/**
+		 * Method to set the FloatingUI options
+		 *
+		 * @memberof Balloon
+		 */
+		public setFloatingUIOptions(): void {
```

Comment:
Don't think so.

Again, FloatingUI is just a collection of utils methods, it creates no Instance. It's not a pattern by itself.
We will never want to have a different provider for the FloatingUI, because FloatingUI **_is the provider_**! 

In the future, if we find a library that does a better job at this, we will create a new Util and use it or replace it internally on the pattern that we see fit.

That's why neither the Balloon or OverflowMenu are ProviderPatterns. FloatingUI is just a function that helps a specific DOM element to position itself on the screen.

Reply:
Will try to implement your suggestion 👍